apps/sms/test/marionette/conversation_test.js

```diff
@@ -491,4 +503,101 @@ marionette('Conversation Panel Tests', function() {
+      // several recipients
+      newMessageView.typeMessage('message2');
+      newMessageView.addNewRecipient('+234');
+
+      newMessageView.showSubject();
+      newMessageView.typeSubject('subject1');
+      conversationView = newMessageView.send();
+
+      newMessageView = conversationView.addRecipients();
+      assert.equal(newMessageView.recipients.length, 2,
+        'There should be 2 recipients');
+      assert.include(newMessageView.recipients, '+123',
+        '+123 should be in the recipients');
+      assert.include(newMessageView.recipients, '+234',
+        '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, '', 'MMS text should be empty');
+    });
+
+    test('Sms being written: message and recipients should be kept',
+    function() {
+      conversationView.typeMessage('message2');
+      var newMessageView = conversationView.addRecipients();
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'message text should be kept');
+    });
+
+    test('Mms being written: all info must be kept', function() {
+      conversationView.typeMessage('message2');
+      // add a subject
+      conversationView.showSubject();
+      conversationView.typeSubject('subject1');
+      var newMessageView = conversationView.addRecipients();
+
+      // assert that we kept all the infos
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject1',
+        'Subject should be kept');
+
+      // let's add a recipients and send the message.
+      // we will have a thread with several recipients
+      newMessageView.addNewRecipient('+234');
+      conversationView = newMessageView.send();
+      // write another MMS
+      conversationView.typeMessage('message3');
+      conversationView.showSubject();
+      conversationView.typeSubject('subject2');
+      // add a blob
+      conversationView.addAttachment().choose('Messages Activity Caller')
+        .pickImage();
+      messagesApp.switchTo();
+
+      // redirect to new message to add recipients
+      newMessageView = conversationView.addRecipients();
+
+      // assert the content
+      assert.equal(newMessageView.recipients.length, 2,
+        'There should be 2 recipients');
+      assert.include(newMessageView.recipients, '+123',
+        '+123 should be in the recipients');
+      assert.include(newMessageView.recipients, '+234',
+        '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, 'message3',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject2',
+        'Subject should be kept');
+      assert.ok(newMessageView.attachments, 'Attachment should be kept');
```

Comment:
Actually, enabling gecko logs for the second idea yields 
```bash 
1449314963991	Marionette	DEBUG	conn1 client <- {"error":"element not visible","message":"Element is not currently visible and may not be manipulated","stacktrace":null}
``` 
when I try to tap on the empty `#messages-container`. So 2 and 3 are related: the height of my viewport is not big enough